aspnetcore/performance/response-compression.md

```diff
@@ -0,0 +1,170 @@
+
+Using the sample application, the client would submit a request with the `Accept-Encoding: custom` header. The middleware will use the custom compression implementation and return the response with a `Content-Encoding: custom` header. The client must be able to decompress the custom encoding in order for a custom compression implementation to work.
+
+[!code-csharp[Main](response-compression/sample/Startup.cs?name=snippet2&highlight=6)]
+
+[!code-csharp[Main](response-compression/sample/CustomCompressionProvider.cs?name=snippet1)]
+
+Submit a request to the sample application with the `Accept-Encoding: custom` header and observe the response headers. The `Vary` and `Content-Encoding` headers are present on the response. The response body (not shown) isn't compressed by the sample, as there is no compression implementation in the `CustomCompressionProvider` class of the sample. However, the sample shows where you would implement such a compression algorithm.
+
+![Fiddler window showing result of a request with the Accept-Encoding header and a value of custom. The Vary and Content-Encoding headers are added to the response.](response-compression/_static/request-custom-compression.png)
+
+### MIME types
+The middleware specifies a default set of MIME types for compression:
+* `text/plain`
+* `text/css`
+* `application/javascript`
+* `text/html`
+* `application/xml`
+* `text/xml`
+* `application/json`
+* `text/json`
+
+You can replace or append MIME types with the Response Compression Middleware options. Note that wildcard MIME types, such as `text/*` are not supported. The sample application adds a MIME type for `image/svg+xml` and will compress and serve the ASP.NET Core banner image (*banner.svg*).
+
+[!code-csharp[Main](response-compression/sample/Startup.cs?name=snippet2&highlight=7)]
+
+### Compression with secure protocol
+Compressed responses over secure connections can be controlled with the `EnableForHttps` option, which is disabled by default. Using compression with dynamically generated pages can lead to security problems such as the [CRIME](https://en.wikipedia.org/wiki/CRIME_(security_exploit)) and [BREACH](https://en.wikipedia.org/wiki/BREACH_(security_exploit)) attacks.
+
+## Adding the Vary header
+When compressing responses based on the `Accept-Encoding` header, there are potentially multiple compressed versions of the response and an uncompressed version. In order to instruct client and proxy caches that multiple versions exist and should be stored, you should always supply a `Vary` header with an `Accept-Encoding` value. The sample application adds a `Vary` header using a method; however, the middleware will be upgraded soon to provide this feature ([Basic Middleware #187](https://github.com/aspnet/BasicMiddleware/issues/187)).
```

Comment:
@spboyer This is a 2.0 feature. Not sure how we want to document upcoming features that will not be back ported to 1.1.